src/py/flwr/common/record/recordset.py

```diff
@@ -15,54 +15,65 @@
 """RecordSet."""
 
 
-from dataclasses import dataclass, field
-from typing import Dict
+from dataclasses import dataclass
+from typing import Callable, Dict, Optional, Type, TypeVar
 
 from .configsrecord import ConfigsRecord
 from .metricsrecord import MetricsRecord
 from .parametersrecord import ParametersRecord
+from .typeddict import TypedDict
+
+T = TypeVar("T")
 
 
 @dataclass
 class RecordSet:
-    """Definition of RecordSet."""
-
-    parameters: Dict[str, ParametersRecord] = field(default_factory=dict)
-    metrics: Dict[str, MetricsRecord] = field(default_factory=dict)
-    configs: Dict[str, ConfigsRecord] = field(default_factory=dict)
-
-    def set_parameters(self, name: str, record: ParametersRecord) -> None:
-        """Add a ParametersRecord."""
-        self.parameters[name] = record
-
-    def get_parameters(self, name: str) -> ParametersRecord:
-        """Get a ParametesRecord."""
-        return self.parameters[name]
+    """Enhanced RecordSet with a unified and Pythonic interface."""
 
-    def del_parameters(self, name: str) -> None:
-        """Delete a ParametersRecord."""
-        del self.parameters[name]
+    _params_records: TypedDict[str, ParametersRecord]
+    _metrics_records: TypedDict[str, MetricsRecord]
+    _configs_records: TypedDict[str, ConfigsRecord]
 
-    def set_metrics(self, name: str, record: MetricsRecord) -> None:
-        """Add a MetricsRecord."""
-        self.metrics[name] = record
+    def __init__(
+        self,
+        parameters_records: Optional[Dict[str, ParametersRecord]] = None,
+        metrics_records: Optional[Dict[str, MetricsRecord]] = None,
+        configs_records: Optional[Dict[str, ConfigsRecord]] = None,
+    ) -> None:
+        def _get_check_fn(__t: Type[T]) -> Callable[[T], None]:
+            def _check_fn(__v: T) -> None:
+                if not isinstance(__v, __t):
+                    raise TypeError(f"Key must be of type str. You passed {type(__v)}.")
```

Comment:
```suggestion
                    raise TypeError(f"Key must be of type `str`, but `{type(__v)}` was passed")
```

Reply:
Also, is `str` correct here? Wouldn't it need to be `__t`?

Reply:
Thanks. I've updated it.